File: problem-solving/parsers/py/_13.py
def parser(f):
	text = f.readline().strip()
	pattern = f.readline().strip()

	assert(f.readline() == "\n")

	cert = int(f.readline())
	assert(f.readline().strip() == "")

	return cert, (text, pattern,)

def verifier(cert, ans):
	return cert == ans

def error(cert, input, ans):
	print("Input:")
	print("text = " + str(input[0]))
	print("pattern = " + str(input[1]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()

File: problem-solving/parsers/java/_13.java
import java.util.Scanner;

public class _13 implements Parser {
    public Object[] parser(Scanner f) {
        String text = f.nextLine();
        String pattern = f.nextLine();

        assert(f.nextLine() == "");

        int cert = f.nextInt();

        return Parser.ret_parser(cert, text, pattern);
    }

    public Object run(Object[] input) {
        return Solutions.numDistinct((String)input[0], (String)input[1]);
    }

    public boolean verifier(Object cert, Object ans) {
        return (int)cert == (int)ans;
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "text = %s\n" +
            "pattern = %s\n" +
            "\n" +
            "Expected:\n" +
            "%d\n" +
            "\n" +
            "Actual:\n" +
            "%d\n",
            (String)input[0],
            (String)input[1],
            (int)cert,
            (int)ans
        ));
    }

}
